dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -0,0 +1,516 @@
+        JsonArray jsonArray = data.Data["Get"]![className]!.AsArray();
+
+        // ReSharper disable once LoopCanBeConvertedToQuery
+        foreach (JsonNode? json in jsonArray)
+        {
+            string id = json!["_additional"]!["id"]!.GetValue<string>();
+            Embedding<float> vector = Embedding<float>.Empty;
+            if (json["_additional"]!["vector"] != null)
+            {
+                IEnumerable<float> floats = json["_additional"]!["vector"]!.AsArray().Select(a => a!.GetValue<float>());
+                vector = new(floats);
+            }
+
+            string text = json["sk_text"]!.GetValue<string>();
+            string description = json["sk_description"]!.GetValue<string>();
+            string additionalMetadata = json["sk_additional_metadata"]!.GetValue<string>();
+            string key = json["sk_id"]!.GetValue<string>();
+            DateTime? timestamp = json["sk_timestamp"] != null
+                ? Convert.ToDateTime(json["sk_timestamp"]!.GetValue<string>(), CultureInfo.InvariantCulture)
+                : null;
+
+            MemoryRecord memoryRecord = MemoryRecord.LocalRecord(
+                id,
+                text,
+                description,
+                vector,
+                additionalMetadata,
+                key,
+                timestamp);
+
+            double distance = json["_additional"]!["distance"]!.GetValue<double>();
+
+            result.Add((memoryRecord, distance));
+        }
+
+        foreach ((MemoryRecord, double) kv in result)
+        {
+            yield return kv;
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<(MemoryRecord, double)?> GetNearestMatchAsync(
+        string collectionName,
+        Embedding<float> embedding,
+        double minRelevanceScore = 0,
+        bool withEmbedding = false,
+        CancellationToken cancellationToken = default)
+    {
+        IAsyncEnumerable<(MemoryRecord, double)> results = this.GetNearestMatchesAsync(
+            collectionName,
+            embedding,
+            minRelevanceScore: minRelevanceScore,
+            limit: 1,
+            withEmbeddings: withEmbedding,
+            cancellationToken: cancellationToken);
+
+        (MemoryRecord, double) record = await results.FirstOrDefaultAsync(cancellationToken).ConfigureAwait(false);
+
+        return (record.Item1, record.Item2);
+    }
+
+    // Get a class description, useful for checking name collisions
+    private static string ToWeaviateFriendlyClassDescription(string collectionName)
+    {
+        return $"Semantic Kernel memory store for collection: {collectionName}";
+    }
+
+    // Convert a collectionName to a valid Weaviate class name
+    private static string ToWeaviateFriendlyClassName(string collectionName)
+    {
+        // Prefix class names with SK to ensure proper case for Weaviate Classes
+        return $"SK{s_classNameRegEx.Replace(collectionName, string.Empty)}";
```

Comment:
Same comment here, not sure why we need `SK` prefix, sorry if I missed something. Any explanation will be highly appreciated!

Reply:
Prefixing the collection name with `SK` at indicates that this is a collection (Weaviate calls these 'class names') that is owned, controlled and managed by Semantic Kernel. In multi-tenant/mixed environments I figured it would be a useful convention to have this in place.

Reply:
Based on discussion within internal team, prefixing is not needed. We will remove it in other vector db implementations as well. 

Answers:
> this is a collection (Weaviate calls these 'class names') that is owned, controlled and managed by Semantic Kernel. This is needed in listing collections.

This is not the case. Collection is always owned, controlled and managed by user. Semantic Kernel just uses it as instrument how to operate with memory. For example, when you create SQL table with Entity Framework, EF doesn't prefix your table with some naming, because it's not needed, unless user wants to mark SQL tables with some prefix explicitly.

> In multi-tenant/mixed environments I figured it would also be a useful convention to have this in place.

Again, user is responsible how to manage multi-tenant/mixed environments, not Semantic Kernel itself.

> The other reason is that Weaviate requires all class names have a leading uppercased letter.

In this case we can convert first letter to uppercase, if I want to create `contacts` collection, we can transform it to `Contacts` and push it to Weaviate, if this is required by API.

Reply:
Prefix removed. Code now ensures name starts with an uppercase letter.